### DISCUSSION TITLE
FormatException: No deserialization found for type dynamic

### DISCUSSION THREAD

**Discussion by realkhalife** [serverpod/serverpod#444] [Q&A]
Hi guys, why the function `session.db.query()` return type is `List<List<dynamic>>`  instead of `List<Map<String,dynamic>>` ?

**Comment by yahu1031** [comment]
Cause as you deep dive into that function [query()](https://pub.dev/documentation/serverpod/latest/database/DatabaseConnection/query.html) > [pgExecutionContext.query(...)](https://pub.dev/documentation/postgres/latest/postgres/PostgreSQLExecutionContext/query.html) > which returns a future [PostgresSQLResult](https://pub.dev/documentation/postgres/latest/postgres/PostgreSQLResult-class.html) which is implemented by [PostgreSQLResultRow](https://pub.dev/documentation/postgres/latest/postgres/PostgreSQLResultRow-class.html) which is implemented by [List](https://api.dart.dev/stable/2.18.4/dart-core/List-class.html).

And as of that nowhere it was implemented a `Map<String, dynamic>` and when you get these kinds of dynamic responses, We better don't explicitly state the type of response instead going with dynamic and casting on need is a better idea.

> NOTE: Please convert this issue to a discussion. @vlidholt 
> Please do use [discussions](https://github.com/serverpod/serverpod/discussions) for questions @realkhalife

**Comment by realkhalife** [comment]
Now, when i run :
```
Future<List<List<dynamic>>> searchPost(Session session, String query) async {
    try {
      return await session.db.query(
          "SELECT name FROM posts WHERE document @@ to_tsquery('$query')");
    } catch (e) {
      rethrow;
    }
  }
```

I got this error message : 

```
I/flutter ( 7000): Failed call: post.searchPost
I/flutter ( 7000): FormatException: No deserialization found for type dynamic
```

here is the Post protocol :

```
class: Post
table : posts

fields:
  uid : String
  postId : String
  name: String 
  description: String?
  mediaUrls: List<String>
  comments : Comment?
  date : DateTime
  rating : Rating?
  isChecked : bool?
```

**Comment by yahu1031** [comment]
Similar to #426, #432

**Comment by realkhalife** [comment]
i tried to follow the steps in #426, i forced dynamic to be String but nothing work.

**Reply by yahu1031** [reply]
Can you please rename the title to `FormatException: No deserialization found for type dynamic`.

This must be treated/fixed with #426 & #432
